test/functions.coffee

```diff
@@ -150,13 +150,13 @@ test "@-parameters and splats with constructors", ->
   eq b, obj.last
 
 test "destructuring in function definition", ->
-  (([{a: [b], c}]...) ->
+  (({a: [b], c}) ->
     eq 1, b
     eq 2, c
   ) {a: [1], c: 2}
 
   context = {}
-  (([{a: [b, c = 2], @d, e = 4}]...) ->
+  (({a: [b, c = 2], @d, e = 4}) ->
```

Comment:
These tests pass with or without this change. I just thought the revised version was a more realistic test. We could test both versions if you want, but the `[]...` syntax is still tested in the `` #4005: `([a = {}]..., b) ->` weirdness `` test, so I’m not sure we need to test it here too.